### ISSUE TITLE
How to define a function of type js.Dynamic => _ ?

### ISSUE THREAD

**Issue by akiradeveloper** [scalajs-react-interface/sri#11]
This is an example code to flip a colored rectangle when the button is pressed but still in error when running on emulator. What's wrong with my code?

```scala
object Experimental {
  case class State(
    angleAnim: AnimatedValue = new AnimatedValue(0)
  )
  class Component extends NavigationScreenComponentS[State] {
    initialState(State())
    var angle: Double = 0
    def render() = {
      import styles._
      val onPressHandler = () => {
        state.angleAnim.setValue(angle)
        Animated.timing(
          this.state.angleAnim,
          new TimingAnimationConfig {
            toValue = angle + 180
            duration = 1000 // ms?
          }
        ).start()
      }
      View(style = whole)(
        AnimatedView(style = tile(angle))(),
        Button(title = "flip", onPress = onPressHandler)
      )
    }
    val callbackFun = (x: js.Dynamic) => {
      // angle = x.asInstanceOf[Double]
    }
    override def componentDidMount(): Unit = {
      state.angleAnim.addListener(callbackFun)
    }
  }
  object styles extends InlineStyleSheetUniversal {
    import dsl._
    val whole = style(
      flex := 1,
      alignItems.center
    )
    def tile(angle: Double) = style(
      width := 100,
      height := 100,
      backgroundColor := (if (90 <= angle && angle < 270) "blue" else "red"),
      transform := js.Array(
        js.Dynamic.literal(rotateX = s"${angle}deg")
      )
    )
  }
}
```

![2018-02-05 12 07 29](https://user-images.githubusercontent.com/785824/35786801-2ac45bb8-0a6d-11e8-94f5-7773cd23b03f.png)

The error says, I guess, the function I passed isn't a function. But it's compiled and I can't see anything wrong.

![2018-02-05 12 01 10](https://user-images.githubusercontent.com/785824/35786778-fd980248-0a6c-11e8-9eea-8dddc3e97c47.png)

Here is what is generated by Scala.js, related to the callback function.

```js
function $s_Le9_default_Experimental$Component__componentDidMount__Le9_default_Experimental$Component__V($this) {
  $as_Le9_default_Experimental$State($this.scalaState).angleAnim$1.addListener($as_F1($this.callbackFun))
}
...
function $s_Le9_default_Experimental$Component__$$anonfun$callbackFun$1__p6__Le9_default_Experimental$Component__sjs_js_Dynamic__V($this, x) {
  /*<skip>*/
}
var $b_Le9_default_Experimental$Component = (void 0);
function $a_Le9_default_Experimental$Component() {
  if ((!$b_Le9_default_Experimental$Component)) {
    /** @constructor */
    var $c_Le9_default_Experimental$Component = (function $c_Le9_default_Experimental$Component() {
      $a_Lsri_navigation_NavigationScreenComponentS().call(this, $m_Lsri_core_package$().neq__Lsri_core_package$$eq$colon$bang$eq());
      Object.defineProperty(this, "angle", {
        "configurable": true,
        "enumerable": true,
        "writable": true,
        "value": 0.0
      });
      Object.defineProperty(this, "callbackFun", {
        "configurable": true,
        "enumerable": true,
        "writable": true,
        "value": null
      });
      this.initialState(new $c_Le9_default_Experimental$State($m_Le9_default_Experimental$State$().apply$default$1__Lsri_universal_apis_AnimatedValue()));
      this.angle = 0.0;
      this.callbackFun = new $c_sjsr_AnonFunction1((function($this) {
        return (function(x$2) {
          var x = x$2;
          $s_Le9_default_Experimental$Component__$$anonfun$callbackFun$1__p6__Le9_default_Experimental$Component__sjs_js_Dynamic__V($this, x)
        })
      })(this))
    });
```

**Comment by chandu0101** [comment]
ah wrong facade def 
```scala
def addListener(callback: js.Dynamic => _): String = js.native
```
Correct Version : 

```scala
def addListener(callback: js.Function1[js.Dynamic,Any]): String = js.native
```

can you send a PR for this and if possible convert all other lambdas to js.FunctionX equivalants in same file.

**Comment by akiradeveloper** [comment]
alright

**Comment by akiradeveloper** [comment]
@chandu0101 Thanks for merging. Btw, how can I test the new universal alone before the entire update and release?

**Comment by chandu0101** [comment]
you can `publishLocal` and update version number for universal in your project `build.sbt` to test in your local env . I am releasing new version now anyway.

**Comment by chandu0101** [comment]
published `2018.2.9-RC`
